cloud/services/groups/groups.go

```diff
@@ -30,7 +30,13 @@ import (
 
 // Get provides information about a resource group.
 func (s *Service) Get(ctx context.Context, spec interface{}) (resources.Group, error) {
-	return s.Client.Get(ctx, s.Scope.ResourceGroup())
+	resourceGroup, err := s.Client.Get(ctx, s.Scope.ResourceGroup())
+	if err != nil && azure.ResourceNotFound(err) {
+		return resources.Group{}, errors.Wrapf(err, "resource group %s not found", s.Scope.ResourceGroup())
+	} else if err != nil {
```

Comment:
we don't need this extra else statement, you could just `return resourceGroup, err`
if err is nil this is equivalent to  `return resourceGroup, nil`

Reply:
thanks
@CecileRobertMichon should we do this for the other services as well?

Reply:
done

Reply:
> should we do this for the other services as well?

Not in this PR, we can get to it later

Reply:
yes yes, in followups. will check other services, @CecileRobertMichon 
thx